llama-index-integrations/llms/llama-index-llms-huggingface-api/llama_index/llms/huggingface_api/base.py

```diff
@@ -136,6 +136,83 @@ def class_name(cls) -> str:
+        description=(
+            "The value of the frequency_penalty parameter to use when generating text."
+            " This parameter is used to penalize words that are already in the text."
+        ),
+    )
+    repetition_penalty: Optional[float] = Field(
+        default=None,
+        description=(
+            "The value of the repetition_penalty parameter to use when generating text."
+            " This parameter is used to penalize words that are repeated in the text."
+        ),
+    )
+    presence_penalty: Optional[float] = Field(
+        default=None,
+        description=(
+            "The value of the presence_penalty parameter to use when generating text."
+            " This parameter is used to penalize words that are not in the text."
+        ),
+    )
+    max_tokens: Optional[int] = Field(
+        default=None,
+        description=(
+            "The maximum number of tokens to generate when generating text. This parameter"
+            " is used to limit the length of the generated text."
+        ),
+    )
+    max_new_tokens: Optional[int] = Field(
+        default=None,
+        description=(
+            "The maximum number of tokens to generate when generating text. This parameter"
+            " is used to limit the length of the generated text."
+        ),
+    )
+    temperature: Optional[float] = Field(
+        default=None,
+        description=(
+            "The value of the temperature parameter to use when generating text."
+            " This parameter is used to control the randomness of the generated text."
+        ),
+    )
+    top_k: Optional[int] = Field(
+        default=None,
+        description=(
+            "The value of the top_k parameter to use when generating text. This parameter"
+            " is used to limit the number of tokens that can be generated."
+        ),
+    )
+    top_p : Optional[float] = Field(
+        default=None,
+        description=(
+            "The value of the top_p parameter to use when generating text. This parameter"
+            " is used to limit the cumulative probability of the generated tokens."
+        ),
+    )
+    top_n_tokens : Optional[int] = Field(
+        default=None,
+        description=(
+            "The value of the top_n_tokens parameter to use when generating text. This parameter"
+            " is used to limit the number of tokens that can be generated."
+        ),
+    )
+    seed: Optional[int] = Field(
+        default=None,
+        description=(
+            "The value of the seed parameter to use when generating text. This parameter"
+            " is used to control the randomness of the generated text."
+        ),
+    )
+    stop: Optional[List[str]] = Field(
+        default=None,
+        description=(
+            "The value of the stop parameter to use when generating text. This parameter"
+            " is used to stop generating text when one of the specified tokens is generated."
+        ),
+    )
```

Comment:
None of these parameters are used though? Seems like there needs to be some function to get model kwargs when using the client?

Reply:
Yes! Sorry for the oversight. Correct me if I am wrong - [link](https://github.com/run-llama/llama_index/blob/main/llama-index-integrations/llms/llama-index-llms-huggingface-api/llama_index/llms/huggingface_api/base.py#L216) - here we can add these extra arguments. I am new to open-source contributions, I would really appreciate some guidance.

Reply:
Hmm, not in  the metadata, but I think these options need to be passed when using the clients, like here
https://github.com/run-llama/llama_index/blob/1f62f0ee6bee5fdb9e4dec95845662f1969be590/llama-index-integrations/llms/llama-index-llms-huggingface-api/llama_index/llms/huggingface_api/base.py#L228

Probably we want a helper function to gather up the kwargs, and then each time a client is used, that function can help gather all the kwargs and send them along

Reply:
Working on it! Thanks !